src/solver.jl

```diff
@@ -2,30 +2,88 @@
 #  This Source Code Form is subject to the terms of the Mozilla Public
 #  License, v. 2.0. If a copy of the MPL was not distributed with this
 #  file, You can obtain one at http://mozilla.org/MPL/2.0/.
-#===================================================
- This file implements the default PajaritoSolver
-===================================================#
-@enum DisaggSOC _disagg_soc_default _disagg_soc_on _disagg_soc_off
+
+#=========================================================
+This file implements the default PajaritoSolver
+=========================================================#
 
 export PajaritoSolver
+
 immutable PajaritoSolver <: MathProgBase.AbstractMathProgSolver
-    verbose::Int                # Verbosity level flag
-    algorithm                   # Choice of algorithm: "OA" or "BC"
-    mip_solver                  # Choice of MILP solver
-    cont_solver                 # Choice of Conic solver
-    opt_tolerance               # Relatice optimality tolerance
-    time_limit                  # Time limit
-    profile::Bool               # Performance profile switch
-    disaggregate_soc::DisaggSOC # SOCP disaggregator for SOC constraints
-    instance::AbstractString    # Path to instance
-    enable_sdp::Bool            # Indicator for enabling sdp support
-    force_primal_cuts::Bool     # Enforces primal cutting planes under conic solver
-    dual_cut_zero_tol::Float64  # Tolerance to check if a conic_dual is zero or not
+    path::AbstractString        # File path
+    log_level::Int              # Verbosity level flag
+    branch_cut::Bool            # Use BC algorithm, else use OA
+    misocp::Bool                # Use SOC/SOCRotated cones in the MIP model (if MIP solver supports MISOCP)
+    disagg::Bool                # Disaggregate SOC/SOCRotated cones in MIP (if solver is conic)
+    drop_dual_infeas::Bool      # Do not add cuts from dual cone infeasible dual vectors
+    proj_dual_infeas::Bool      # Project dual cone infeasible dual vectors onto dual cone boundaries
+    proj_dual_feas::Bool        # Project dual cone strictly feasible dual vectors onto dual cone boundaries
+    solver_mip::MathProgBase.AbstractMathProgSolver # MIP solver
+    solver_con::MathProgBase.AbstractMathProgSolver # Continuous solver
+    timeout::Float64            # Time limit for OA/BC not including initial load
```

Comment:
i figured it feels more natural to do everything in seconds, including printouts of time taken etc. it's easy for the user to write `timeout = 5*60*60` for 5 hours, or 5*60 for 5 minutes. using different units in different places might be more confusing